EIPS/eip-2935.md

```diff
@@ -103,21 +111,32 @@ push1 0x20
 push0
 return
 
-// revert
+// 0x53: revert
 jumpdest
 push0
 push0
 revert
 
+// 0x57: set op - sstore the input to number-1
+jumpdest
+push0
+calldataload
+push1 0x1
+number
+sub
+sstore
+
 stop
 ```
 
-Note that the input contract read `32` bytes input as `calldataload`. Users and clients doing EVM call to this contract should left pad the `arg` correctly.
+Corresponding bytecode:
+`3373fffffffffffffffffffffffffffffffffffffffe1460575767ffffffffffffffff5f3511605357600143035f3511604b575f35612000014311604b57611fff5f3516545f5260205ff35b5f5f5260205ff35b5f5ffd5b5f35600143035500`
 
-<!-- TODO: bytecode is based off on first version and will be updated once assembly is locked down as it changes contract sender and address -->
+#### Contract `get` and `set` mechanism
 
-Corresponding bytecode:
-`60203611603157600143035f35116029575f356120000143116029576120005f3506545f5260205ff35b5f5f5260205ff35b5f5ffd00`
+Similar to [EIP-4788](./eip-4788.md) above contract provides a `set` mechanism which clients can choose to update the block's parent hash in the ring buffer instead of direct state update. For this clients should call this contract with `SYSTEM_ADDRESS` as caller and provide `32` bytes parent block hash as input.
```

Comment:
```suggestion
Similar to [EIP-4788](./eip-4788.md) above contract provides a `set` mechanism which clients can choose to update the block's parent hash in the ring buffer instead of direct state update. For this clients should call this contract with `SYSTEM_ADDRESS` as caller and provide `32` bytes parent block hash as input.

Note that the inputs to the `set` method are not sanitized and it is the caller's responsibility to ensure that the ring buffer isn't overrun. This choice was made to support a change in the ring buffer size without having to redeploy a new contract.
```

Reply:
we will also need to change the `get` method as well. but there is another consideration: 

the contracts that would directly call the contract would break if we modify ring buffer length and expcted them to mod the input in a new way